horace_core/sqw/@sqw/tobyfit_DGfermi_resconv.m

```diff
@@ -138,11 +187,41 @@
 k_to_v = lookup.k_to_v;
 k_to_e = lookup.k_to_e;
 
+% Create pointers to parts of lookup structure for the current dataset
+x0 = lookup.x0{iw};
+xa = lookup.xa{iw};
+x1 = lookup.x1{iw};
+thetam = lookup.thetam{iw};
+angvel = lookup.angvel{iw};
+ki = lookup.ki{iw};
+kf = lookup.kf{iw}(idx_start:idx_end, :);
+s_mat = lookup.s_mat{iw};
+spec_to_rlu = lookup.spec_to_rlu{iw};
+alatt = lookup.alatt{iw};
+angdeg = lookup.angdeg{iw};
+is_mosaic = lookup.is_mosaic{iw};
+dt = lookup.dt{iw}(idx_start:idx_end, :);
+en = lookup.en{iw}(idx_start:idx_end, :);
+
+% Run and detector for each pixel
+npix = idx_end - idx_start + 1; %sqw_obj.pix.num_pixels;
+[irun, idet] = parse_pixel_indices(sqw_obj, (idx_start:idx_end)');     % returns column vectors
+
+% Get detector information for each pixel in the sqw object
+% size(x2) = [npix,1], size(d_mat) = [3,3,npix], size(f_mat) = [3,3,npix]
+% and size(detdcn) = [3,npix]
+[x2, detdcn, d_mat, f_mat] = detector_table.func_eval_ind (iw, irun, idet, @detector_info);
+
+% Catch case of refining crystal orientation
+if refine_crystal
+    % Strip out crystal refinement parameters and reorient datasets
+    [sqw_obj, pars{1}] = refine_crystal_strip_pars (sqw_obj, xtal, pars{1});
+
+    % Update s_mat, spec_to_rlu, alatt and angdeg because in general the
+    % crystal orientation and lattice parameters will have changed
+    [~,s_mat,spec_to_rlu,alatt,angdeg]=sample_coords_to_spec_to_rlu(sqw_obj.experiment_info);
 
-% Perform resolution broadening calculation
-% -----------------------------------------
-% Package parameters as a cell for convenience
-if ~iscell(pars), pars={pars}; end
+end
 
 % Catch case of refining moderator parameters
 if refine_moderator
```

Comment:
The if refine_moderator ...end block only needs to be done once in the call to tobyfit_DGfermi_resconv. This block can be moved to immediately before line 139. Like the case of if refine_crystal ... end  I think it affects performance, as well as logically belonging earlier as it is repeating the same assignment for each chunk of each sqw object.